pkg/config/remote/configs_apm_sampling.go

```diff
@@ -54,3 +52,15 @@ func (c *apmSamplingConfigs) update(configFiles map[string]configFiles) (*APMSam
 	}
 	return nil, nil
 }
+
+func (c *apmSamplingConfigs) shouldUpdate(configFiles map[string]configFiles) bool {
+	if c.config == nil || len(c.config.Configs) != len(configFiles) {
```

Comment:
Update cases:

1. No remote configs in the agent
2. New config id is added
3. An existing config id is removed
4. One of the config ids has a newer version

Reply:
What happens when shouldUpdate is false? Does the empty configs make it to pkg/trace/remote_sampler and erase existing configs?